docs/STYLE.md

```diff
@@ -0,0 +1,232 @@
+
+We always write our enums and structs as `typedef`s. (Usually one can't use an enum typedef as a function argument since enum typedefs are implicitly `s32`.)
+
+## Naming
+
+Names are "big-endian": the most significant/largest part of the system goes first, e.g. `DM_RAVINE_STATE_ACTIVE` rather than `DM_RAVINE_ACTIVE_STATE`.
+
+| Type                 | Style                   | Example                 |
+| -------------------- | ----------------------- | ----------------------- |
+| Local variables      | camelCase               | `yawToPlayer`           |
+| Global variables     | gCamelCase              | `gSaveContext`          |
+| Static variables[^1] | sCamelCase              | `sZeroVec`              |
+| Struct members       | camelCase               | `actionFunc`            |
+| Struct names         | PascalCase              | `EnFirefly`             |
+| Enum types           | PascalCase              | `EnFireflyMainType`     |
+| Enum values          | SCREAMING_SNAKE_CASE    | `AT_ON`                 |
+| Defines/macros       | SCREAMING_SNAKE_CASE    | `SCREEN_WIDTH`,`ABS(x)` |
+| Functions            | SystemName_FunctionName | `Actor_SpawnAsChild`    |
+| Files                | snake_case              | `z_en_firefly.c`        |
+
+[^1]: including in-function static
+
+Action functions are usually named with a simple present-tense verb or verb phrase: `{...}_Talk`, `{...}_Wait`, `{...}_FallToGround`, etc. Setup functions are `Setup{name of action}`.
+
+Ideally names should be both short and clear, although it's better to be clear than short.
+
+## Formatting
+
+A lot of formatting is done by clang-format, such as
+
+- indent is 4 spaces, tabs are not used
+- case labels indented
+- 120 column limit
+- brackets go on the same line (`if (1) {`)
+- pointer goes on type (`s32* var;` not `s32 *var;`)
+
+There are various other conventions that it does not catch, though:
+
+- Blank line between declarations and code:
+  ```c
+  s32 var;
+  
+  func();
+  ```
+- combine declarations and definitions if possible:
+  ```c
+  s32 var = 0;
+  
+  func();
+  ```
+  instead of
+  ```c
+  s32 var;
+  
+  var = 0;
+  func();
+  ```
+- blank lines between switch cases if they're long (use your judgement).
+
+
+## Numbers
+
+### dec(imal)
+
+- timers
+- colours and alpha
+- Usually array accesses and sizes
+
+### hex(adecimal)
+
+- angles (for now; the code itself is very inconsistent with round hex, round dec, and degrees)
+- Addresses
+- Bitmasks (i.e. `& 0x80` etc.)
+- Struct offset comments
+
+Numbers below `10`/`0xA` do not need the `0x` if by themselves in code.
+
+### Booleans
+
+If a function returns only `0` or `1`, and is used as a boolean (i.e. in conditionals), replace the returns by `false` and `true`. (We do not use `bool`, partly because is a C99 thing, and partly because the original has used almost any integer type as a boolean return at some point!)
```

Comment:
```suggestion
If a function returns only `0` or `1`, and is used as a boolean (i.e. in conditionals), replace the returns by `false` and `true`. (We do not use `bool`, partly because is a C99 thing, and partly because the original has used almost every integer type as a boolean return at some point!)
```